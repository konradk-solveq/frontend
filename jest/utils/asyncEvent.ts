import {act} from '@testing-library/react-native';

const asyncEvent = async (event: any) => {
    await act(async () => {
        await event;
    });
};

export default asyncEvent;
