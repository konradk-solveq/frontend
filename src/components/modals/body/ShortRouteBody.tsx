import React from 'react';
import {useMergedTranslation} from '@utils/translations/useMergedTranslation';
import {BodyPrimary, Header3} from '@src/components/texts/texts';
import {getFVerticalPx} from '@src/theme/utils/appLayoutDimensions';

const boldedText = (string: string) => {
    const stringToBold = string.split('**');

    if (stringToBold?.[1]) {
        return (
            <>
                {stringToBold?.[0] || ''}
                <Header3>{stringToBold?.[1]}</Header3>
                {stringToBold?.[2] || ''}
            </>
        );
    }

    return string;
};

interface IProps {
    showAlterMessage?: string;
}

const ShortRouteBody: React.FC<IProps> = ({showAlterMessage}: IProps) => {
    const {t} = useMergedTranslation('CounterThankYouPage');

    return (
        <>
            {!showAlterMessage ? (
                <>
                    <BodyPrimary>
                        {boldedText(t('errorMessageTitle'))}
                    </BodyPrimary>
                    <BodyPrimary style={{marginTop: getFVerticalPx(16)}}>
                        {t('errorMessageContent')}
                    </BodyPrimary>
                </>
            ) : (
                <BodyPrimary>{showAlterMessage}</BodyPrimary>
            )}
        </>
    );
};

export default ShortRouteBody;
