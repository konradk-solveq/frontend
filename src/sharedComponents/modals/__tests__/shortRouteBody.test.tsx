import 'react-native';
import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import {initAppSize} from '@helpers/layoutFoo';

import ShortRouteBody from '@sharedComponents/modals/body/shortRouteBody';
import asyncEvent from '@jestUtils/asyncEvent';

/**
 * It seems that function fireEvent.press doesn't respect
 * TouchableOpacity with prop disabled from react-native-gesture-handler,
 * so default platform is set to 'android'
 */
jest.mock('react-native/Libraries/Utilities/Platform', () => ({
    OS: 'android',
    select: () => null,
}));

describe('<ShortRouteBody />', () => {
    describe('Rendering', () => {
        beforeAll(() => {
            initAppSize();
        });

        it('Should match snapshot', async () => {
            const onCloseFun = jest.fn();

            const component = await asyncEvent(
                render(<ShortRouteBody onClose={onCloseFun} />),
            );
            expect(component).toMatchSnapshot();
        });

        it('Should render component with alternative message', async () => {
            const onCloseFun = jest.fn();
            const alterMessage = 'Other message to show';

            const component = await asyncEvent(
                render(
                    <ShortRouteBody
                        onClose={onCloseFun}
                        showAlterMessage={alterMessage}
                    />,
                ),
            );

            const withMessage = component.getByText(alterMessage);
            expect(withMessage).toBeTruthy();

            expect(component).toMatchSnapshot();
        });

        it('Should render component with loader on btn', async () => {
            const onCloseFun = jest.fn();
            const testID = 'red-btn-loader';

            const component = await asyncEvent(
                render(<ShortRouteBody onClose={onCloseFun} withBtnLoader />),
            );

            const withLoader = component.getByTestId(testID);
            expect(withLoader).toBeTruthy();

            expect(component).toMatchSnapshot();
        });

        it('Should render component with disabled btn', async () => {
            const onCloseFun = jest.fn();
            const testID = 'big-red-btn-for-short-route';

            const component = await render(
                <ShortRouteBody onClose={onCloseFun} withBtnLoader />,
            );

            const disabledBtn = component.getByTestId(testID);
            await asyncEvent(fireEvent.press(disabledBtn));
            expect(onCloseFun).not.toHaveBeenCalled();

            expect(component).toMatchSnapshot();
        });

        afterEach(() => {
            jest.resetModules();
        });
    });
});
