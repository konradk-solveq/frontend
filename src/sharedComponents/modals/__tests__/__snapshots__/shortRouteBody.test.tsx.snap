// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortRouteBody /> Rendering Should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
      "paddingBottom": 215.88169642857144,
      "width": "100%",
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#d8232a",
              "fontFamily": "DIN2014Narrow-Regular",
              "fontSize": 40,
              "left": 72.46376811594203,
              "paddingVertical": 5,
              "textAlign": "center",
              "width": 605.072463768116,
            },
            Object {
              "marginTop": 205.45982142857144,
            },
          ]
        }
      >
        Coś poszło nie tak!
      </Text>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginTop": 44.66517857142858,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={536.231884057971}
          bbWidth={753.6231884057971}
          focusable={false}
          height={536.231884057971}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 536,
                "width": 753,
              },
            ]
          }
          vbHeight={296}
          vbWidth={416}
          width={753.6231884057971}
        >
          <RNSVGGroup>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "height": 536.231884057971,
                  },
                  Object {
                    "width": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <RNCWebView
                    cacheEnabled={true}
                    injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
                    injectedJavaScriptForMainFrameOnly={true}
                    javaScriptEnabled={true}
                    messagingEnabled={false}
                    onContentProcessDidTerminate={[Function]}
                    onHttpError={[Function]}
                    onLoadingError={[Function]}
                    onLoadingFinish={[Function]}
                    onLoadingProgress={[Function]}
                    onLoadingStart={[Function]}
                    onMessage={[Function]}
                    onShouldStartLoadWithRequest={[Function]}
                    scalesPageToFit={true}
                    scrollEnabled={false}
                    showsHorizontalScrollIndicator={false}
                    showsVerticalScrollIndicator={false}
                    source={
                      Object {
                        "html": "<!DOCTYPE html><html lang=\\"pl-PL\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html;  charset=utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0\\" /><style>html,body,svg {margin:0;padding:0;height:100%;width:100%;overflow:hidden;background-color:transparent} svg{position:fixed}</style></head><body><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 416 296\\" width=\\"412\\" height=\\"296\\">
        <path d=\\"M1.7 132.2c-4.3 31.4 11.5 45.6 47.6 42.5 54-4.6 52 113.4 147.6 84.4 34.6-10.5 51-100.1 110-95.7 88.3 6.7 60.6-110 108.1-82.6\\" stroke=\\"#D8232A\\" fill=\\"none\\" 
        fill-rule=\\"evenodd\\" stroke-dasharray=\\"6 9\\" stroke-linecap=\\"round\\">
            <animate attributeName=\\"stroke-dashoffset\\" dur=\\"100s\\" repeatCount=\\"indefinite\\" values=\\"0 ; -1000\\"/>
        </path>
    </svg></body></html>",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                        },
                        Object {
                          "backgroundColor": "transparent",
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                    useSharedProcessPool={true}
                    useWebKit={false}
                  />
                </View>
              </View>
            </View>
            <RNSVGGroup>
              <RNSVGPath
                d="M202.7 0h10.6C292.1 0 356 66.3 356 148s-63.9 148-142.7 148h-10.6C123.9 296 60 229.7 60 148S123.9 0 202.7 0z"
                fill={-622973}
                fillOpacity={0.2}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
              <RNSVGPath
                d="m251.6 71 34.4 34.4L165.4 226 131 191.6z"
                fill={-1}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={-2612438}
                strokeLinecap={1}
                strokeLinejoin={1}
              />
              <RNSVGPath
                d="m243 105-13-13m-13 39-13-13m-13 39-13-13m-13 39-13-13"
                propList={
                  Array [
                    "stroke",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={-2612438}
                strokeLinecap={1}
                strokeLinejoin={1}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        style={
          Object {
            "color": "#414141",
            "fontFamily": "DIN2014Narrow-Light",
            "fontSize": 23,
            "left": 72.46376811594203,
            "marginTop": 29.776785714285715,
            "width": 605.072463768116,
          }
        }
      >
        Przejechany dystans jest 
        <Text
          style={
            Object {
              "fontFamily": "DIN2014Narrow-Bold",
              "fontSize": 23,
            }
          }
        >
          zbyt krótki
        </Text>
        .
      </Text>
      <Text
        style={
          Object {
            "color": "#414141",
            "fontFamily": "DIN2014Narrow-Light",
            "fontSize": 23,
            "left": 72.46376811594203,
            "marginTop": 29.776785714285715,
            "width": 605.072463768116,
          }
        }
      >
        Nagraj dłuższą trasę (minimalna długość trasy powinna być większa niż 100 metrów).
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 96.77455357142858,
        "left": 72.46376811594203,
        "position": "absolute",
        "width": 605.072463768116,
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
          "marginTop": 44.66517857142858,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d8232a",
            "borderRadius": 50,
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": "100%",
          }
        }
        testID="big-red-btn-for-short-route"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "DIN2014Narrow-Bold",
                "fontSize": 20,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          ZAMKNIJ
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ShortRouteBody /> Rendering Should render component with alternative message 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
      "paddingBottom": 215.88169642857144,
      "width": "100%",
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#d8232a",
              "fontFamily": "DIN2014Narrow-Regular",
              "fontSize": 40,
              "left": 72.46376811594203,
              "paddingVertical": 5,
              "textAlign": "center",
              "width": 605.072463768116,
            },
            Object {
              "marginTop": 205.45982142857144,
            },
          ]
        }
      >
        Coś poszło nie tak!
      </Text>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginTop": 44.66517857142858,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={536.231884057971}
          bbWidth={753.6231884057971}
          focusable={false}
          height={536.231884057971}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 536,
                "width": 753,
              },
            ]
          }
          vbHeight={296}
          vbWidth={416}
          width={753.6231884057971}
        >
          <RNSVGGroup>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "height": 536.231884057971,
                  },
                  Object {
                    "width": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <RNCWebView
                    cacheEnabled={true}
                    injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
                    injectedJavaScriptForMainFrameOnly={true}
                    javaScriptEnabled={true}
                    messagingEnabled={false}
                    onContentProcessDidTerminate={[Function]}
                    onHttpError={[Function]}
                    onLoadingError={[Function]}
                    onLoadingFinish={[Function]}
                    onLoadingProgress={[Function]}
                    onLoadingStart={[Function]}
                    onMessage={[Function]}
                    onShouldStartLoadWithRequest={[Function]}
                    scalesPageToFit={true}
                    scrollEnabled={false}
                    showsHorizontalScrollIndicator={false}
                    showsVerticalScrollIndicator={false}
                    source={
                      Object {
                        "html": "<!DOCTYPE html><html lang=\\"pl-PL\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html;  charset=utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0\\" /><style>html,body,svg {margin:0;padding:0;height:100%;width:100%;overflow:hidden;background-color:transparent} svg{position:fixed}</style></head><body><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 416 296\\" width=\\"412\\" height=\\"296\\">
        <path d=\\"M1.7 132.2c-4.3 31.4 11.5 45.6 47.6 42.5 54-4.6 52 113.4 147.6 84.4 34.6-10.5 51-100.1 110-95.7 88.3 6.7 60.6-110 108.1-82.6\\" stroke=\\"#D8232A\\" fill=\\"none\\" 
        fill-rule=\\"evenodd\\" stroke-dasharray=\\"6 9\\" stroke-linecap=\\"round\\">
            <animate attributeName=\\"stroke-dashoffset\\" dur=\\"100s\\" repeatCount=\\"indefinite\\" values=\\"0 ; -1000\\"/>
        </path>
    </svg></body></html>",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                        },
                        Object {
                          "backgroundColor": "transparent",
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                    useSharedProcessPool={true}
                    useWebKit={false}
                  />
                </View>
              </View>
            </View>
            <RNSVGGroup>
              <RNSVGPath
                d="M202.7 0h10.6C292.1 0 356 66.3 356 148s-63.9 148-142.7 148h-10.6C123.9 296 60 229.7 60 148S123.9 0 202.7 0z"
                fill={-622973}
                fillOpacity={0.2}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
              <RNSVGPath
                d="m251.6 71 34.4 34.4L165.4 226 131 191.6z"
                fill={-1}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={-2612438}
                strokeLinecap={1}
                strokeLinejoin={1}
              />
              <RNSVGPath
                d="m243 105-13-13m-13 39-13-13m-13 39-13-13m-13 39-13-13"
                propList={
                  Array [
                    "stroke",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={-2612438}
                strokeLinecap={1}
                strokeLinejoin={1}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        style={
          Object {
            "color": "#414141",
            "fontFamily": "DIN2014Narrow-Light",
            "fontSize": 23,
            "left": 72.46376811594203,
            "marginTop": 29.776785714285715,
            "width": 605.072463768116,
          }
        }
      >
        Other message to show
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 96.77455357142858,
        "left": 72.46376811594203,
        "position": "absolute",
        "width": 605.072463768116,
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
          "marginTop": 44.66517857142858,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#d8232a",
            "borderRadius": 50,
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "opacity": Object {
              "_animation": null,
              "_children": Array [],
              "_listeners": Object {},
              "_offset": 0,
              "_startingValue": 1,
              "_value": 1,
            },
            "width": "100%",
          }
        }
        testID="big-red-btn-for-short-route"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "DIN2014Narrow-Bold",
                "fontSize": 20,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          ZAMKNIJ
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<ShortRouteBody /> Rendering Should render component with disabled btn 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
      "paddingBottom": 215.88169642857144,
      "width": "100%",
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#d8232a",
              "fontFamily": "DIN2014Narrow-Regular",
              "fontSize": 40,
              "left": 72.46376811594203,
              "paddingVertical": 5,
              "textAlign": "center",
              "width": 605.072463768116,
            },
            Object {
              "marginTop": 205.45982142857144,
            },
          ]
        }
      >
        Coś poszło nie tak!
      </Text>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginTop": 44.66517857142858,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={536.231884057971}
          bbWidth={753.6231884057971}
          focusable={false}
          height={536.231884057971}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 536,
                "width": 753,
              },
            ]
          }
          vbHeight={296}
          vbWidth={416}
          width={753.6231884057971}
        >
          <RNSVGGroup>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "height": 536.231884057971,
                  },
                  Object {
                    "width": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <RNCWebView
                    cacheEnabled={true}
                    injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
                    injectedJavaScriptForMainFrameOnly={true}
                    javaScriptEnabled={true}
                    messagingEnabled={false}
                    onContentProcessDidTerminate={[Function]}
                    onHttpError={[Function]}
                    onLoadingError={[Function]}
                    onLoadingFinish={[Function]}
                    onLoadingProgress={[Function]}
                    onLoadingStart={[Function]}
                    onMessage={[Function]}
                    onShouldStartLoadWithRequest={[Function]}
                    scalesPageToFit={true}
                    scrollEnabled={false}
                    showsHorizontalScrollIndicator={false}
                    showsVerticalScrollIndicator={false}
                    source={
                      Object {
                        "html": "<!DOCTYPE html><html lang=\\"pl-PL\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html;  charset=utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0\\" /><style>html,body,svg {margin:0;padding:0;height:100%;width:100%;overflow:hidden;background-color:transparent} svg{position:fixed}</style></head><body><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 416 296\\" width=\\"412\\" height=\\"296\\">
        <path d=\\"M1.7 132.2c-4.3 31.4 11.5 45.6 47.6 42.5 54-4.6 52 113.4 147.6 84.4 34.6-10.5 51-100.1 110-95.7 88.3 6.7 60.6-110 108.1-82.6\\" stroke=\\"#D8232A\\" fill=\\"none\\" 
        fill-rule=\\"evenodd\\" stroke-dasharray=\\"6 9\\" stroke-linecap=\\"round\\">
            <animate attributeName=\\"stroke-dashoffset\\" dur=\\"100s\\" repeatCount=\\"indefinite\\" values=\\"0 ; -1000\\"/>
        </path>
    </svg></body></html>",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                        },
                        Object {
                          "backgroundColor": "transparent",
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                    useSharedProcessPool={true}
                    useWebKit={false}
                  />
                </View>
              </View>
            </View>
            <RNSVGGroup>
              <RNSVGPath
                d="M202.7 0h10.6C292.1 0 356 66.3 356 148s-63.9 148-142.7 148h-10.6C123.9 296 60 229.7 60 148S123.9 0 202.7 0z"
                fill={-622973}
                fillOpacity={0.2}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
              <RNSVGPath
                d="m251.6 71 34.4 34.4L165.4 226 131 191.6z"
                fill={-1}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={-2612438}
                strokeLinecap={1}
                strokeLinejoin={1}
              />
              <RNSVGPath
                d="m243 105-13-13m-13 39-13-13m-13 39-13-13m-13 39-13-13"
                propList={
                  Array [
                    "stroke",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={-2612438}
                strokeLinecap={1}
                strokeLinejoin={1}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        style={
          Object {
            "color": "#414141",
            "fontFamily": "DIN2014Narrow-Light",
            "fontSize": 23,
            "left": 72.46376811594203,
            "marginTop": 29.776785714285715,
            "width": 605.072463768116,
          }
        }
      >
        Przejechany dystans jest 
        <Text
          style={
            Object {
              "fontFamily": "DIN2014Narrow-Bold",
              "fontSize": 23,
            }
          }
        >
          zbyt krótki
        </Text>
        .
      </Text>
      <Text
        style={
          Object {
            "color": "#414141",
            "fontFamily": "DIN2014Narrow-Light",
            "fontSize": 23,
            "left": 72.46376811594203,
            "marginTop": 29.776785714285715,
            "width": 605.072463768116,
          }
        }
      >
        Nagraj dłuższą trasę (minimalna długość trasy powinna być większa niż 100 metrów).
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 96.77455357142858,
        "left": 72.46376811594203,
        "position": "absolute",
        "width": 605.072463768116,
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
          "marginTop": 44.66517857142858,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#db4f54",
            "borderRadius": 50,
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "opacity": Object {
              "_animation": null,
              "_children": Array [],
              "_listeners": Object {},
              "_offset": 0,
              "_startingValue": 1,
              "_value": 1,
            },
            "width": "100%",
          }
        }
        testID="big-red-btn-for-short-route"
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="small"
          testID="red-btn-loader"
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`<ShortRouteBody /> Rendering Should render component with loader on btn 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "height": "100%",
      "paddingBottom": 215.88169642857144,
      "width": "100%",
    }
  }
>
  <RCTScrollView>
    <View>
      <Text
        style={
          Array [
            Object {
              "color": "#d8232a",
              "fontFamily": "DIN2014Narrow-Regular",
              "fontSize": 40,
              "left": 72.46376811594203,
              "paddingVertical": 5,
              "textAlign": "center",
              "width": 605.072463768116,
            },
            Object {
              "marginTop": 205.45982142857144,
            },
          ]
        }
      >
        Coś poszło nie tak!
      </Text>
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "marginTop": 44.66517857142858,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={536.231884057971}
          bbWidth={753.6231884057971}
          focusable={false}
          height={536.231884057971}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 536,
                "width": 753,
              },
            ]
          }
          vbHeight={296}
          vbWidth={416}
          width={753.6231884057971}
        >
          <RNSVGGroup>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "height": 536.231884057971,
                  },
                  Object {
                    "width": undefined,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      undefined,
                    ]
                  }
                >
                  <RNCWebView
                    cacheEnabled={true}
                    injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
                    injectedJavaScriptForMainFrameOnly={true}
                    javaScriptEnabled={true}
                    messagingEnabled={false}
                    onContentProcessDidTerminate={[Function]}
                    onHttpError={[Function]}
                    onLoadingError={[Function]}
                    onLoadingFinish={[Function]}
                    onLoadingProgress={[Function]}
                    onLoadingStart={[Function]}
                    onMessage={[Function]}
                    onShouldStartLoadWithRequest={[Function]}
                    scalesPageToFit={true}
                    scrollEnabled={false}
                    showsHorizontalScrollIndicator={false}
                    showsVerticalScrollIndicator={false}
                    source={
                      Object {
                        "html": "<!DOCTYPE html><html lang=\\"pl-PL\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html;  charset=utf-8\\"><meta name=\\"viewport\\" content=\\"width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0\\" /><style>html,body,svg {margin:0;padding:0;height:100%;width:100%;overflow:hidden;background-color:transparent} svg{position:fixed}</style></head><body><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 416 296\\" width=\\"412\\" height=\\"296\\">
        <path d=\\"M1.7 132.2c-4.3 31.4 11.5 45.6 47.6 42.5 54-4.6 52 113.4 147.6 84.4 34.6-10.5 51-100.1 110-95.7 88.3 6.7 60.6-110 108.1-82.6\\" stroke=\\"#D8232A\\" fill=\\"none\\" 
        fill-rule=\\"evenodd\\" stroke-dasharray=\\"6 9\\" stroke-linecap=\\"round\\">
            <animate attributeName=\\"stroke-dashoffset\\" dur=\\"100s\\" repeatCount=\\"indefinite\\" values=\\"0 ; -1000\\"/>
        </path>
    </svg></body></html>",
                      }
                    }
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Object {
                          "backgroundColor": "#ffffff",
                        },
                        Object {
                          "backgroundColor": "transparent",
                          "height": "100%",
                          "width": "100%",
                        },
                      ]
                    }
                    useSharedProcessPool={true}
                    useWebKit={false}
                  />
                </View>
              </View>
            </View>
            <RNSVGGroup>
              <RNSVGPath
                d="M202.7 0h10.6C292.1 0 356 66.3 356 148s-63.9 148-142.7 148h-10.6C123.9 296 60 229.7 60 148S123.9 0 202.7 0z"
                fill={-622973}
                fillOpacity={0.2}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
              <RNSVGPath
                d="m251.6 71 34.4 34.4L165.4 226 131 191.6z"
                fill={-1}
                propList={
                  Array [
                    "fill",
                    "stroke",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={-2612438}
                strokeLinecap={1}
                strokeLinejoin={1}
              />
              <RNSVGPath
                d="m243 105-13-13m-13 39-13-13m-13 39-13-13m-13 39-13-13"
                propList={
                  Array [
                    "stroke",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={-2612438}
                strokeLinecap={1}
                strokeLinejoin={1}
              />
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        style={
          Object {
            "color": "#414141",
            "fontFamily": "DIN2014Narrow-Light",
            "fontSize": 23,
            "left": 72.46376811594203,
            "marginTop": 29.776785714285715,
            "width": 605.072463768116,
          }
        }
      >
        Przejechany dystans jest 
        <Text
          style={
            Object {
              "fontFamily": "DIN2014Narrow-Bold",
              "fontSize": 23,
            }
          }
        >
          zbyt krótki
        </Text>
        .
      </Text>
      <Text
        style={
          Object {
            "color": "#414141",
            "fontFamily": "DIN2014Narrow-Light",
            "fontSize": 23,
            "left": 72.46376811594203,
            "marginTop": 29.776785714285715,
            "width": 605.072463768116,
          }
        }
      >
        Nagraj dłuższą trasę (minimalna długość trasy powinna być większa niż 100 metrów).
      </Text>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "bottom": 96.77455357142858,
        "left": 72.46376811594203,
        "position": "absolute",
        "width": 605.072463768116,
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
          "marginTop": 44.66517857142858,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#db4f54",
            "borderRadius": 50,
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "opacity": Object {
              "_animation": null,
              "_children": Array [],
              "_listeners": Object {},
              "_offset": 0,
              "_startingValue": 1,
              "_value": 1,
            },
            "width": "100%",
          }
        }
        testID="big-red-btn-for-short-route"
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="small"
          testID="red-btn-loader"
        />
      </View>
    </View>
  </View>
</View>
`;
